lib/dynamoid/transaction_write/action.rb

```diff
@@ -37,6 +42,30 @@ def range_key
         attributes[model_class.range_key]
       end
 
+      # sets a value in the attributes
+      def set(values)
+        attributes.merge!(values)
+      end
+
+      # adds to array of fields for use in REMOVE update expression
+      def remove(field)
+        removals << field
+      end
+
+      # increments a number or adds to a set, starts at 0 or [] if it doesn't yet exist
+      def add(values)
+        additions.merge!(values)
+      end
+
+      # deletes a value or values from a set type or simply sets a field to nil
+      def delete(field_or_values)
+        if field_or_values.is_a?(Hash)
+          deletions.merge!(field_or_values)
+        else
+          remove(field_or_values)
+        end
+      end
```

Comment:
Currently there is no `remove` operation in `Document#update` method. `delete` removes an attribute if no value is specified and deletes collection elements otherwise. It has benefits to keep this abstraction and don't expose to user what DynamoDB parameters are used - legacy AttributeUpdates or more flexible UpdateExpression.

I propose to have only `#delete` method and don't expose `#remove` public method (and don't test it directly - only by calling the `#delete` method)

Reply:
Agreed. I wasn't intending for remove() to be used publicly. I removed it.